app/models/policy/single_bid_auction.rb

```diff
@@ -0,0 +1,110 @@
+class Policy::SingleBidAuction < Policy::Auction
+  TYPE_CODE = 'single_bid'.freeze
+
+  # disallow bids if the user has already placed a bid in addition to
+  # general bidding rules
+  def user_can_bid?
+    super && !user_is_bidder?
+  end
+
+  def bids?
+    return user_is_bidder? if available?
+    super
+  end
+
```

Comment:
Use of super is sometimes considered a smell. Usually there is a way to add another method to encapsulate the original behavior that super is doing. For example:

```
class Policy::Auction
  def can_user_bid?
    some_other_concept
  end

  def some_other_concept
    # original logic from can_user_bid?
  end

  # ...
end

class Policy::SingleBidAuction < Policy::Auction
  # ...
  def user_can_bid?
    some_other_concept && !user_is_bidder?
  end
  # ...
end
```

The main point is not to avoid super as much as to name missing concepts